services/datamanager/builtin/builtin.go

```diff
@@ -610,28 +614,82 @@ func (svc *builtIn) Reconfigure(
 	if svc.syncSensor != syncSensor {
 		svc.syncSensor = syncSensor
 	}
-
 	syncConfigUpdated := svc.syncDisabled != svcConfig.ScheduledSyncDisabled || svc.syncIntervalMins != svcConfig.SyncIntervalMins ||
 		!reflect.DeepEqual(svc.tags, svcConfig.Tags) || svc.fileLastModifiedMillis != fileLastModifiedMillis ||
 		svc.maxSyncThreads != newMaxSyncThreadValue
 
 	if syncConfigUpdated {
+		svc.syncConfigUpdated = syncConfigUpdated
 		svc.syncDisabled = svcConfig.ScheduledSyncDisabled
 		svc.syncIntervalMins = svcConfig.SyncIntervalMins
 		svc.tags = svcConfig.Tags
 		svc.fileLastModifiedMillis = fileLastModifiedMillis
 		svc.maxSyncThreads = newMaxSyncThreadValue
+	}
+
+	// if datacapture is enabled, kick off a go routine to check if disk space is filling due to
+	// cached datacapture files
+	if !svc.captureDisabled {
+		fileDeletionCtx, cancelFunc := context.WithCancel(context.Background())
+		svc.fileDeletionRoutineCancelFn = cancelFunc
+		svc.fileDeletionBackgroundWorkers = &sync.WaitGroup{}
+		svc.fileDeletionBackgroundWorkers.Add(1)
+		go pollFilesystem(fileDeletionCtx, svc.fileDeletionBackgroundWorkers,
+			svc.captureDir, deleteEveryNthValue, svc.syncer, svc.logger)
+	}
+
+	g.Success()
+	return nil
+}
+
+func (svc *builtIn) startPropagateDataSyncConfig() {
+	svc.dataCaptureWorkers.Add(1)
+	goutils.ManagedGo(svc.propagateDataSyncConfigLoop, svc.dataCaptureWorkers.Done)
+}
 
+// propagateDataSyncConfigLoop runs until Close() is called on *builtIn
+// Immediately on first execution and every second afterwards it
+// checks if the datasync configuration has changes which
+// have not propagated to datasync.
```

Comment:
Just to confirm: the reason this could happen is if the robot was offline when we tried to propagate these changes last time?

Reply:
This happens any time Reconfigure changes any of these config values: https://github.com/viamrobotics/rdk/pull/4164/files#diff-ffb86a3e2b364cc470f33937bb1ffb08ca2ef9d01b2734320daea7bee24a8943R621-R628